File: go/raindrops/raindrops.go
package raindrops

import "strconv"

const testVersion = 3

func Convert(x int) string {
	rain := ""

	if x%3 == 0 {
		rain += "Pling"
	}
	if x%5 == 0 {
		rain += "Plang"
	}
	if x%7 == 0 {
		rain += "Plong"
	}
	if rain == "" {
		rain = strconv.Itoa(x)
	}

	return rain
}

File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(drops)
    sound = ""

    if drops % 3 == 0
      sound += "Pling"
    end
    if drops % 5 == 0
      sound += "Plang"
    end
    if drops % 7 == 0
      sound += "Plong"
    end

    if sound == ""
      sound = drops.to_s
    end

    return sound
  end
end

class BookKeeping
  VERSION = 3
end
